metadata/reader.go

```diff
@@ -582,37 +595,39 @@ func attrFromZtocEntry(src *ztoc.FileMetadata, dst *Attr) *Attr {
 	return dst
 }
 
-func getIDByName(md map[uint32]*metadataEntry, name string, rootID uint32) (uint32, error) {
-	name = cleanEntryName(name)
-	if name == "" {
+// getIDByName returns the node ID associated with a path (file or directory). It recurses
+// the path all the way back to the root node, returning every child ID along the way.
+func getIDByName(md map[uint32]*metadataEntry, path string, rootID uint32) (uint32, error) {
+	if path == "" {
 		return rootID, nil
 	}
-	dir, base := filepath.Split(name)
-	pid, err := getIDByName(md, dir, rootID)
+	parentDirectory, base := filepath.Split(filepath.Clean(path))
+	parentID, err := getIDByName(md, parentDirectory, rootID)
```

Comment:
Yes, we clean the current path before recursing. When we reach a direct child of a root node, eg: `a/` we clean the path so it becomes `a` and `filepath.Split` on a path with no separators gives you an empty parent/directory => `""`.

Reply:
Can you add to the test cases a directory of `/foo/`? My read is this would result in infinite recursion because clean would yield `/foo` followed by split yielding `parent = /` and `base = foo`

Reply:
My concern is because we have removed the `TrimSuffix` from `parentDir` func, then input like "/foo/bar/" would result in infinite recursion.

Reply:
Edit: ah I see the trim prefix is in the clean path function. That covers my concern. Thanks.